homeassistant/components/media_player/androidtv.py

```diff
@@ -0,0 +1,413 @@
+        hass.data[DATA_KEY] = {}
+
+    from adb.client import Client as AdbClient
+    client = AdbClient(host="127.0.0.1", port=5037)
+
+    androidtvs = []
+    for host, device_config in config[CONF_DEVICES].items():
+        name = device_config[CONF_NAME]
+        port = str(device_config[CONF_PORT])
+        scan_interval = device_config[CONF_SCAN_INTERVAL]
+        uri = host + ':' + port
+
+        try:
+            ADBdev = client.device(uri)
+            if ADBdev is None:
+                _LOGGER.error(
+                    "Can't reach adb server, is it running ?")
+                raise PlatformNotReady
+
+            device = {'name': name,
+                      'ipaddr': host,
+                      'port': port,
+                      'scan_interval': scan_interval}
+            androidtv = AndroidTv(device)
+            androidtvs.append(androidtv)
+            hass.data[DATA_KEY][host] = androidtv
+        except RuntimeError:
+            _LOGGER.error(
+                "Can't reach adb server, is it running ?")
+            raise PlatformNotReady
+
+    add_entities(androidtvs, True)
+
+    def service(service):
+        """Dispatch service calls to target entities."""
+        params = {key: value for key, value in service.data.items()
+                  if key != ATTR_ENTITY_ID}
+
+        entity_ids = service.data.get(ATTR_ENTITY_ID)
+        if entity_ids:
+            target_devices = [dev for dev in hass.data[DATA_KEY].values()
+                              if dev.entity_id in entity_ids]
+        else:
+            target_devices = hass.data[DATA_KEY].values()
+
+        for target_device in target_devices:
+            if len(params.keys()) == 1:
+                if 'action' in params:
+                    target_device.do_action(params['action'])
+                elif 'intent' in params:
+                    target_device.start_intent(params['intent'])
+                elif 'key' in params:
+                    target_device.input_key(params['key'])
+
+    hass.services.register(DOMAIN, 'androidtv_action', service)
+
+
+class AndroidTv(MediaPlayerDevice):
+    """Representation of an AndroidTv device."""
+
+    def __init__(self, device):
+        """Initialize the Android device."""
+        from adb.client import Client as AdbClient
+        host = device['ipaddr']
+        port = device['port']
+        name = device['name']
+        scan_interval = device['scan_interval']
+        self._scan_interval = scan_interval
+
+        self._uri = host + ':' + port
+        self._client = AdbClient(host="127.0.0.1", port=5037)
+        self._device = self._client.device(self._uri)
+
+        self._name = name
+        self._available = None
+        self._volume = None
+        self._muted = None
+        self._state = None
+        self._app_id = None
+        self._app_name = None
+        self._device = None
+        _LOGGER.info("its all good man")
+
+    def _init_regular_updates(self):
```

Comment:
This isn't used and we don't need it. This entity is a polling entity so it will call `update` by default at a regular interval. That interval can be set to a default value via defining the constant `SCAN_INTERVAL`, as has been done. This will be overridable by the user via `CONF_SCAN_INTERVAL` in the platform config schema by default.